src/m365/spfx/commands/project/project-github-workflow-add.ts

```diff
@@ -0,0 +1,203 @@
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+
+        if (args.options.scope && args.options.scope === 'sitecollection') {
+          if (!args.options.siteUrl) {
+            return `siteUrl option has to be defined when scope set to ${args.options.scope}`;
+          }
+
+          const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.siteUrl);
+          if (isValidSharePointUrl !== true) {
+            return isValidSharePointUrl;
+          }
+        }
+
+        if (args.options.loginMethod && SpfxProjectGithubWorkflowAddCommand.loginMethod.indexOf(args.options.loginMethod) < 0) {
+          return `${args.options.loginMethod} is not a valid login method. Allowed values are ${SpfxProjectGithubWorkflowAddCommand.loginMethod.join('|')}`;
+        }
+
+        if (args.options.scope && SpfxProjectGithubWorkflowAddCommand.scope.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpfxProjectGithubWorkflowAddCommand.scope.join('|')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
+    if (this.projectRootPath === null) {
+      throw new CommandError(`Couldn't find project root folder`, SpfxProjectGithubWorkflowAddCommand.ERROR_NO_PROJECT_ROOT_FOLDER);
+    }
+
+    const solution: string = path.join(this.projectRootPath as string, 'sharepoint', 'solution');
+    if (!fs.existsSync(solution)) {
+      throw new CommandError(`No SPFx app package found`);
+    }
+
+    if (this.debug) {
+      logger.logToStderr(`Adding GitHub workflow in the current SPFx project`);
+    }
+
+    try {
+      this.updateWorkflow(solution, workflow, args.options);
+      this.saveWorkflow(workflow);
+    }
+    catch (error: any) {
+      throw new CommandError(error);
+    }
+  }
+
+  private saveWorkflow(workflow: any): void {
+    const githubPath: string = path.join(this.projectRootPath as string, '.github');
+    this.ensureDirectory(githubPath);
+
+    const workflowPath: string = path.join(githubPath, 'workflows');
+    this.ensureDirectory(workflowPath);
+
+    const workflowFile: string = path.join(workflowPath, 'deploy-spfx-solution.yml');
+    fs.writeFileSync(path.resolve(workflowFile), yaml.stringify(workflow), 'utf-8');
+  }
+
+  private updateWorkflow(solution: string, workflow: any, options: GlobalOptions): void {
+    let sppkgFileName: string = fs.readdirSync(solution).filter(fn => fn.endsWith('.sppkg'))[0];
+    sppkgFileName = sppkgFileName.replace('.sppkg', '');
+
+    workflow.name = workflow.name.replace('{{ name }}', options.name ?? sppkgFileName);
+
+    if (options.branchName) {
+      workflow.on.push.branches[0] = options.branchName;
+    }
+
+    if (options.manuallyTrigger) {
+      workflow.on['workflow_dispatch'] = null;
```

Comment:
Why are we using here `workflow.on['workflow_dispatch']` instead of `workflow.on.workflow_dispatch`? The latter is typesafe, and we should use it instead. Let's use the same pattern in other places where we use indexers

Reply:
yes for sure. Mainly I did that because we have a eslint rule preventing me from doing this
![image](https://github.com/pnp/cli-microsoft365/assets/58668583/5b8dedf8-92c4-436b-bab4-a6df1c81aa53)

do you think it will be better to disable it for this line?

Reply:
I change to what you suggested and disabled eslint for this line. I also updated similar cases below